src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/fallbackerrorhandler.cpp

```diff
@@ -70,16 +74,16 @@ void FallbackErrorHandler::error(const LogString& message,
     }
 }
 
-void FallbackErrorHandler::setAppender(const AppenderPtr& primary1) {
+void FallbackErrorHandler::setAppender(const AppenderWeakPtr& primary1) {
     LogLog::debug(((LogString) LOG4CXX_STR("FB: Setting primary appender to ["))
-                  + primary1->getName() + LOG4CXX_STR("]."));
-    this->primary = primary1;
+                  + primary1.lock()->getName() + LOG4CXX_STR("]."));
+    primary_ = primary1;
```

Comment:
@SNiukalov  maybe after you renamed `primary -> primary_` it's time to rename `primary1 -> primary`? or `primary1` have some special meaning. And the same below with `backup1`